README.md

```diff
@@ -1,2 +1,94 @@
-# C++ libp2p implementation
-page is loading, please wait...
+# CPP-Libp2p  
+
+> Fully compatible C++17 implementation of libp2p library
+
+Libp2p is a modular networking stack described in [spec](https://github.com/libp2p/specs)
+
+## Dependencies
+
+All dependencies are managed using [Hunter](hunter.sh). It uses cmake to download required libraries and do not require to download and install packages manually.
+Target C++ compilers are:
+* GCC 7.4
+* Clang 6.0.1
+* AppleClang 11.0
+
+## Supported protocols
+* Transports: TCP
+* Security protocols: [Plaintext 2.0](https://github.com/libp2p/specs/blob/master/plaintext/README.md), [SECIO](https://github.com/libp2p/specs/blob/master/secio/README.md) (WIP)
+* Multiplexing protocols: [MPlex](https://github.com/libp2p/specs/tree/master/mplex), [Yamux](https://github.com/hashicorp/yamux/blob/master/spec.md)
+* [Kademlia DHT](https://github.com/libp2p/specs/pull/108) (WIP)
+* [Identify](https://github.com/libp2p/specs/tree/master/identify)
+
+## Development
+### Clone
+
+To clone repository execute
+```
+git clone https://github.com/soramitsu/libp2p.git
+```
+
+### Build cpp-libp2p
+
+First build will likely take long time. However, you can cache binaries to [hunter-binary-cache](https://github.com/soramitsu/hunter-binary-cache) or even download binaries from the cache in case someone has already compiled project with the same compiler. To this end, you need to set up two environment variables:
+```
+GITHUB_HUNTER_USERNAME=<github account name>
+GITHUB_HUNTER_TOKEN=<github token>
+```
+To generate github token follow the [instructions](https://help.github.com/en/github/authenticating-to-github/creating-a-personal-access-token-for-the-command-line). Make sure `read:packages` and `write:packages` permissions are granted (step 7 in instructions).
+
+This project is can be built with
+
+```
+mkdir build && cd build
+cmake -DCLANG_TIDY=ON ..
+make -j
+```
+
+It is suggested to build project with clang-tidy checks, however if you wish to omit clang-tidy step, you can use `cmake ..` instead.
+
+Tests can be run with: 
+```
+cd build
+ctest
+```
+
+### CodeStyle
+
+We follow [CppCoreGuidelines](https://github.com/isocpp/CppCoreGuidelines).
+
+Please use provided [.clang-format](.clang-format) file to autoformat the code.
+
+## Examples
+
+Please follow [example](example) section to read examples of how to use the library
```

Comment:
```suggestion
Please explore [example](example) subdirectory to study examples of how to use the library
```

section -> subdirectory since we are already reading examples section of readme